### ISSUE TITLE
regarding dynamic service/action support implementation

### ISSUE THREAD

**Issue by mnumanuyar** [osrf/dynamic_message_introspection#24]
i am aware there is a cloed issue for it already (#21) but [this line](https://github.com/osrf/dynamic_message_introspection/blob/50277e341f61455c42a6015b5718804a4f8588f6/dynmsg/src/typesupport.cpp#L173) hardcodes "msg" part of the function name.

 My solution is adding a new parameter to get_type_info func, with a default value of "msg3" to not break anything. However maybe changes to "InterfaceTypeName" makes more sense. 

As a side note [here](https://github.com/osrf/dynamic_message_introspection/pull/23#issuecomment-1707279977) says maybe this is repo is not fully maintained. 

To sum up my questions:
- Is this repo maintained? if not, is there a maintained version of this repo? 
- Is supporting actions/services in scope of this repo? 
- Do you have a suggestion regarding solution? Would it be better to change InterfaceTypeName instead of  get_type_info?

**Comment by mnumanuyar** [comment]
fyi, changeing function def is enough to run something like test_dynmsg/examples/conversion_cpp.cpp. So converting to yaml and back works with just this. but i didnt test it troughly otherwise

And also thanks for fantastic package :D
